### ISSUE TITLE
[java] UnnecessaryBooleanAssertion overlaps with SimplifiableTestAssertion

### ISSUE THREAD

**Issue by testation21** [pmd/pmd#3087]
**Affects PMD Version:** 6.30

**Rule:** UnnecessaryBooleanAssertion
https://pmd.github.io/pmd-6.30.0/pmd_rules_java_errorprone.html#unnecessarybooleanassertion

**Description:** There is a problem when the boolean value is used with `!` after a function call to test the function.
Please, check this case.

Thank you.

**Code Sample demonstrating the issue:**

```java
import org.junit.Assert;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

class TestUnnecessaryBooleanAssertion {
        @Test
        public void testPMD(){
                test(foo);
        }

        protected void test(Foo foo) {
                boolean testResult = testFunction(foo);

                assertTrue(!testResult);    //an alarm is generated
                assertTrue(testResult);    //an alarm isn't generated
    }
}

```

**Expected outcome:**
Don't show any alarms.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
I think, this is a problem in the rule documentation. `assertTrue(!testResult)` may be replaced by `assertFalse(testResult)`, and the rule is explicitly written to catch this (see [tests](https://github.com/pmd/pmd/blob/df80080aaa2d48f3edd3e88cc3348e57925e0d4e/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/UnnecessaryBooleanAssertion.xml#L56)). This is however not documented.

Relates to #2908

**Comment by YoonhoChoi96** [comment]
Oh, I got it.
Thanks!

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
